sdk/tests/sdk/whirlpools/whirlpool-impl.test.ts

```diff
@@ -495,4 +495,217 @@ describe("whirlpool-impl", () => {
+      pool
+    );
+
+    const tickLowerArrayData = await ctx.fetcher.getTickArray(
+      positionWithFees.tickArrayLower,
+      true
+    );
+
+    const tickUpperArrayData = await ctx.fetcher.getTickArray(
+      positionWithFees.tickArrayUpper,
+      true
+    );
+
+    const tickLower = TickArrayUtil.getTickFromArray(
+      tickLowerArrayData!,
+      tickLowerIndex,
+      tickSpacing
+    );
+
+    const tickUpper = TickArrayUtil.getTickFromArray(
+      tickUpperArrayData!,
+      tickUpperIndex,
+      tickSpacing
+    );
+
+    const feesQuote = collectFeesQuote({
+      whirlpool: poolData,
+      position: positionData,
+      tickLower,
+      tickUpper,
+    });
+
+    const dWalletTokenBAccount = await deriveATA(otherWallet.publicKey, poolData.tokenMintB);
+    const rewardAccount0 = await deriveATA(otherWallet.publicKey, poolData.rewardInfos[0].mint);
+    const rewardAccount1 = await deriveATA(otherWallet.publicKey, poolData.rewardInfos[1].mint);
+    const rewardAccount2 = await deriveATA(otherWallet.publicKey, poolData.rewardInfos[2].mint);
+
+    const txs = await pool.closePosition(
+      positionWithFees.publicKey,
+      new Percentage(new u64(10), new u64(100)),
+      otherWallet.publicKey,
+      otherWallet.publicKey,
+      ctx.wallet.publicKey
+    );
+
+    let ataTx: TransactionBuilder | undefined;
+    let closeTx: TransactionBuilder;
+    if (txs.length === 1) {
+      closeTx = txs[0];
+    } else if (txs.length === 2) {
+      ataTx = txs[0];
+      closeTx = txs[1];
+    } else {
+      throw new Error(`Invalid length for txs ${txs}`);
+    }
+
+    const otherWalletBalanceBefore = await ctx.connection.getBalance(otherWallet.publicKey);
+    const positionAccountBalance = await ctx.connection.getBalance(positionWithFees.publicKey);
+
+    await ataTx?.buildAndExecute();
+    await closeTx.addSigner(otherWallet).buildAndExecute();
+
+    const otherWalletBalanceAfter = await ctx.connection.getBalance(otherWallet.publicKey);
+
+    const minAccountExempt = await ctx.fetcher.getAccountRentExempt();
+    const solReceived = otherWalletBalanceAfter - otherWalletBalanceBefore;
+
+    // TODO: Why is the reward quote only correct when data is collected after the tx is ran?
+    const rewardsQuote = collectRewardsQuote({
+      whirlpool: await pool.refreshData(),
+      position: await position.refreshData(),
+      tickLower,
+      tickUpper,
+    });
+    /**
+     * Expected tokenA (SOL) returns on other wallet
+     * 1. withdraw value from decrease_liq (decrease_quote, though not always accurate)
+     * 2. accrrued fees from trade (fee_quote)
+     * 3. Position PDA account rent return (balance from position address account)
+     * 4. wSOL rent-exemption close (getAccountExemption)
+     * 5. Position token account rent return (getAccountExemption)
+     *
+     * Other costs from payer, but not received by other wallet
+     * 1. close_position tx cost
+     * 2. ATA account initialization
+     */
```

Comment:
@atamari-orca - you were close!
The values missing were the rent collected from the closing of the Position PDA & the Position token ATA on the user's wallet.